### DISCUSSION TITLE
Crediting L1 deposits properly in the PP

### DISCUSSION THREAD

**Discussion by bpfarmer** [agglayer/agglayer#153] [Pessimistic-proof]
Right now, L1 deposits are handled separately in the Global Exit Tree. The GET has the following structure (from [here](https://docs.polygon.technology/zkEVM/architecture/protocol/unified-LxLy/ulxly-exit-trees-node/#new-global-exit-tree)): 

![](https://docs.polygon.technology/img/zkEVM/ulxly-mainnet-and-rollup-exit-trees.png)

I think, but I want to verify, that we can handle this in the following way. 

Assume that L1 has a `NetworkID` of 0 - hopefully @krlosMata can confirm.

From the perspective of claiming an `ImportedBridgeExit`, we treat L1->L2 deposits in the same way that we treat an L2->L2 transfer. We import a Mainnet `LocalExitRoot` in the PP and verify the Merkle path of an `ImportedBridgeExit` in this local exit root for mainnet deposits.

The only difference is that when we go to verify the inclusion of `LocalExitRoots` in a `GlobalExitRoot`, we need to special-case the inclusion proof for `NetworkId` = 0. We verify that the root of the `LocalExitRoot` is the left-child of the root of `GlobalExitTree`. This should be sufficient to prove that deposits from mainnet to a particular chain are valid.

**Comment by jesusholoo** [comment]
i cofirm L1 networkID of L1 is 0 yes.
THe first rollup ( Rollup ID = 1) will have netowrkID = 1 as well ( they are coherent ^^⁾)

And also yes, the only difference is exactly that, i have a kinda similar logic on the [Bridge Smart contract](https://github.com/0xPolygonHermez/zkevm-contracts/blob/main/contracts/v2/PolygonZkEVMBridgeV2.sol#L781) to claim Mainnet assets/messages

Also i think you already saw it but, in the "right" side, so, for proving rollups we need an extra merkle proof ( one for the LocalExitRoot of the rollup, and another for the RollupsExitRoot), Mainnet local exit root goes "directly" into the GER

And this design was done this way in order to avoid synch issues between mainnet and the rest of networks, in case we move the computation of `RollupsExitRoot` inside the proof ^^
